src/rust/engine/process_executor/src/main.rs

```diff
@@ -241,8 +274,8 @@ fn main() {
     argv,
     env,
     input_files,
-    output_files: BTreeSet::new(),
-    output_directories: BTreeSet::new(),
+    output_files: output_file_path,
```

Comment:
If you call these variables `output_files` and `output_directories` instead of `output_file_path` and `output_directory_path`, you can omit the `: output_file_path` part of this (like with `input_files` above), which is probably worth doing here because `output_files` and `output_directories` are completely meaningful names :)